cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -811,6 +813,55 @@ def _validate_sparsity(self):
                         f"to use this type of matrix for the given sparsity."
                     )
 
+    def _validate_seurat_convertibility(self):
+        """
+        Use length of component matrices to determine if the anndata object will be unable to be converted to Seurat by
+        virtue of the R language's array size limit (4-byte signed int length). Add warning for each matrix which is
+        too large.
+
+        rtype: None
+        """
+
+        to_validate = [self.adata.X]
+        to_validate_name = ["X"]
+
+        # check if there's raw data
+        if self.adata.raw:
+            to_validate.append(self.adata.raw.X)
+            to_validate_name.append("raw")
+
+        failing_matrix_names: List[str] = []
+        # Check length of component arrays
+        for x, x_name in zip(to_validate, to_validate_name):
+
+            if (
+                    isinstance(x, sparse.csc_matrix)
+                    or isinstance(x, sparse.csr_matrix)
+                    or isinstance(x, sparse.coo_matrix)
+            ):
+                effective_r_array_size = x.count_nonzero()
+
+            elif isinstance(x, ndarray):
+                effective_r_array_size = max(x.shape)
+
+            else:
+                self.warnings.append(
+                    f"{x_name} matrix is of type {type(x)}, sparsity calculation hasn't been "
+                    f"implemented"
+                )
+                continue
+
+            if effective_r_array_size > self._seurat_array_max_length:
+                failing_matrix_names.append(x_name)
+
+        if failing_matrix_names:
+            printable_matrix_names = "\n".join(failing_matrix_names)
+            self.warnings.append(
+                f"The following matrices will fail conversion to Seurat due to exceeding the R array size limit: "
```

Comment:
If this is the warning message, then it does not meet the description requested by the curators:

The `cellxgene-schema` CLI MUST display this warning:

_This dataset cannot be converted to the .rds (Seurat v3) format. **fill-in-the-blank** nonzero elements in **fill-in-the-blank-matrix-name** exceed the limitations in the R dgCMatrix sparse matrix class (2147483647  nonzero elements)._